Yoast/wordpress-seo
# Move migrations dir to config (rename)

## Context


Alternative to #15008 

* Fixes a bug where running the migrations could cause fatal errors on sites where plugins are installed remotely. Some remote plugin installers don't remove files that are removed in an update. Since our migration runner goes through all migrations on the server, this could lead to fatal errors.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where running the migrations could cause fatal errors on sites where plugins are installed remotely. Some remote plugin installers don't remove files that are removed in an update. Since our migration runner goes through all migrations on the server, this could lead to fatal errors.

## Relevant technical choices:

* This one time: rename the migrations dir by moving it to `src/config`. Seems like a better location anyway.
* From now on: never remove a migration again that was once in a release. Instead, empty the `up` and `down` so the migration simply does nothing.

## Test instructions

This PR can be tested by following these steps:

* Install Yoast SEO 14.0.2
* Reset the indexables using Yoast test helper.
* Include [this file](https://github.com/Yoast/wordpress-seo/blob/9.2/migrations/20180515101200_WpYoastIndexableMeta.php) in your migrations directory.
* Get a fatal on the upgrade routine which triggers when you reload.
* Checkout this PR
* Reset the indexables using Yoast test helper.
* Create a migrations folder in the root and include [this file](https://github.com/Yoast/wordpress-seo/blob/9.2/migrations/20180515101200_WpYoastIndexableMeta.php) in your migrations directory.
* Confirm the error is gone when you reload.

## Documentation

## Quality assurance


Fixes #
